### ISSUE TITLE
[Sponsored Transactions] Add support for Shinami sponsorship

### ISSUE THREAD

**Issue by StefPler** [MystenLabs/Sui_Owned_Object_Pools#128]
Depends on issue #125 & should be merged into PR #126.

The Shinami sponsorship API returns the `Signature` separately from the `txBytes` so its not possible to construct a `SignatureWithBytes` object that is required as an output of the `sponsorLambda` anonymous function in the current [execute](https://github.com/MystenLabs/Sui_Owned_Object_Pools/pull/126/files#diff-546bb7085ac912128ea16e94503587026caddf8eb0e5d306403eb6bf9c991d70R79) method of the esHandler.

To support this use case, the 2nd return value of the sponsorLambda function can return either a `Signature` or a `SignatureWithBytes` so its compatible with the current approach while supporting Shinami sponsorship. Might even be desirable to do that to the first argument as well.
